### ISSUE TITLE
MRU isn't selected immedately on tab close.

### ISSUE THREAD

**Issue by VulumeCode** [sblask/webextension-tab-deque#89]
### New Issue Checklist (tick off like this: `[x]`)

- [ x] I checked existing issues (open and closed) for possible duplicates
- [ x] I can reproduce the problem on the latest stable version (not nightly!)

### What is your browser?

- [ x] Firefox
- [ ] Chrome

### What is your operating system?

- [ ] Linux
- [ ] Mac
- [ x] Windows

### Description (please include examples/screenshots where applicable)

When closing a tab, the MRU tab is selected only after Firefox switches to the tab it would normally select. This could trigger some things on that page (and it messes with my new extension). It is possible to switch immediately, as this plugin does https://addons.mozilla.org/en-US/firefox/addon/mru-tab-succession/?utm_source=addons.mozilla.org&utm_medium=referral&utm_content=search

**Comment by sblask** [comment]
The API that that extension uses is cool stuff, but it's not cross browser and I am trying to keep things compatible as much as possible.

**Comment by VulumeCode** [comment]
Then I'd like to add an option to turn MRU-on-close off. While it is in your description, I don't think this is like how windows stack or any of my IDE's tabs work. What do you think?

**Comment by sblask** [comment]
That is the whole point of TabDeque, if you turn it off, what's left? You basically fall back to the browser default behaviour?And that definitely does not behave like a normal window stack. If I have 5 tabs, A B C D E and I select A C E B D and then close them currently selected one after the other the order I expect is D B E C A. That's how my window manager works and what TabDeque implements but not at all how Chrome works. Because of this difference there can be issues.

**Comment by VulumeCode** [comment]
Ahah, for me the main reason to install this add-on was to have the hotkeys to cycle through the deque. At work I do a lot of configuration in webapp interfaces. It's always the same back and forth, but on different groups of tabs all the time. That's also how my IDE back/forward hotkeys work, which I use when I'm exploring code. 

I don't expect MRU behavior when I'm closing tabs, cause then I'm just cleaning up. 

This is just my way of thinking, not saying it is the right way :)

**Comment by sblask** [comment]
Then you can just use the normal shortcuts for switching between tabs, `ctrl-tab` and `ctrl-shift-tab`? You just need to put the tabs next to each other. This could be helpful: https://chrome.google.com/webstore/detail/open-tabs-next-to-current/gmpnnmonpnnmnhpdldahlekfofigiffh?hl=en-GB

**Comment by VulumeCode** [comment]
I guess want best of both worlds :) Sometimes I want MRU, sometimes I want tab bar order.

> You just need to put the tabs next to each other. This could be helpful: https://chrome.google.com/webstore/detail/open-tabs-next-to-current/gmpnnmonpnnmnhpdldahlekfofigiffh?hl=en-GB

But maybe this would make that unnecessary, I'll give it a try! 

Either ways, I'm just making suggestions here. It's your addon in the end, you decide. I'll still make a pull request for the thing from the other thread.

**Comment by kafene** [comment]
@VulumeCode I am running a personal fork of this in which I have used `tabs.moveInSuccession` to get the desired effect.

I haven't published it in a repo, but here is [the gist of it](https://gist.github.com/kafene/1ad8cc4c60129deb3c4f13b78919d8da)

**Comment by sblask** [comment]
I added moveInsSuccession support, already released for Firefox. The API is still not supported in Chrome though.
